server/sockets/sendMessage.js

```diff
@@ -26,19 +22,27 @@ module.exports = (socket) => {
             'Hmmm. It seems your login session has expired. ' +
             'Re-login and try again',
         });
+        return;
+      }
+
+      // Check the message count for the user
+      const userMessageCount = messageCounts[senderId] || 0;
 
+      if (userMessageCount >= 25) {
+        // User has exceeded the message limit
+        socket.emit(NEW_EVENT_SEND_FAILED, {
+          message:
+            'You have exceeded the message limit. Please try again later.',
+        });
         return;
       }
 
-      /**
-       * Cache the sent message in memory a nd persist to db
-       */
+      // Cache the sent message in memory and persist to the database
       const sentMessage = await addMessage(chatId, {
         message,
         time,
         senderId,
         type: 'message',
-        containsBadword
```

Comment:
you can't just remove an important code when merging a conflict check what both code does and don't just delete anything